docs/dev/compiler-toolchain/solidity.md

```diff
@@ -0,0 +1,199 @@
+- `settings/outputSelection/def`
+- `settings/modelChecker`
+
+Additional zkEVM data inserted by `zksolc`:
+
+- `long_version`: the full `solc` version output
+- `zk_version`: the version of the `zksolc` itself
+- `contract/hash`: the hash of the zkEVM bytecode
+- `contract/factory_dependencies`: bytecode hashes of contracts `CREATE`d in the current one. [More detail here](../building-on-zksync/contracts/contract-deployment.md#note-on-factory-deps)
+
+Unsupported sections of the output JSON, ignored by `zksolc`:
+
+- `contracts/<file>/<contract>/evm/bytecode`: replaced by the zkEVM object with only relevant data
+- `contracts/<file>/<contract>/ewasm`
+
+See the complete standard JSON data structures in [the zksolc repository](https://github.com/matter-labs/era-compiler-solidity/tree/main/src/solc/standard_json).
+
+#### Combined JSON
+
+The `zksolc` combined JSON I/O workflow closely follows that of the official `solc` compiler. However, `zksolc` does not
+support a significant part of it, as it only makes sense for compiling for EVM, and has no representation in the zkEVM
+architecture.
+
+The combined JSON is a mere output format, and there is no input combined JSON format. Instead, the CLI arguments are
+used for configuration.
+
+Additional zkEVM data, inserted by `zksolc`:
+
+- `zk_version`: the version of the `zksolc` itself
+- `contract/factory_deps`: bytecode hashes of contracts `CREATE`d in the current one. [More detail here](../building-on-zksync/contracts/contract-deployment.md#note-on-factory-deps).
+
+Unsupported combined JSON flags, rejected by `zksolc`:
+
+- `function-debug`
+- `function-debug-runtime`
+- `generated-sources`
+- `generated-sources-runtime`
+- `opcodes`
+- `srcmap`
+- `srcmap-runtime`
+
+See the complete combined JSON data structures in [the zksolc repository](https://github.com/matter-labs/era-compiler-solidity/tree/main/src/solc/combined_json)
+
+#### Free-form output
+
+This output format is utilized in Yul and LLVM IR compilation modes. These modes only support compiling a single file at
+the moment. Only `--asm` and `--bin` output flags are supported, so this mode can be useful for debugging and prototyping.
+
+#### CLI reference
+
+- `--version`  
+  Print the version and exit.  
+
+- `<input_files>`  
+  Specify the input file paths.  
+  Multiple Solidity files can be passed in the default Solidity mode.  
+  Yul and LLVM IR modes currently support only a single file.  
+
+- `--base-path <path>`  
+  Set the given path as the root of the source tree instead of the root of the filesystem.  
+  Passed to `solc` without changes.  
+
+- `--include-path <path>`  
+  Make an additional source directory available to the default import callback. Can be used multiple times. Can only be used if the base path has a non-empty value.  
+  Passed to `solc` without changes.  
+
+- `--allow-paths <path1,path2,...>`  
+  Allow a given path for imports. A list of paths can be supplied by separating them with a comma.  
```

Comment:
```suggestion
  Allow a given path for imports. Supply a list of paths by separating them with a comma.  
```

Reply:
The same issue with `can be` here.
Sounds like supplying a list is required. A single path is also fine and more common. The comma separation is only needed if you are passing more than one.
Just adding a meaning of `if required` looks good to me.

Reply:
yes, i'm keen to remove all passive tense ... updated suggestion ..